model.py

```diff
@@ -19,8 +19,16 @@ def __init__(self, n_samples: int):
         self.lstm2 = nn.LSTMCell(128, 128)
         self.class_head = nn.Linear(128, 9)
         
-    def stop_criterion(self, tree: list) -> bool:
+    def stop_criterion(self, node_list: list) -> bool:
```

Comment:
Take tree: SyntaxNode, as input, and then we can skip building the tree in lines 24-25.